File: src/my_first_cpp_pkg/src/tests_topics/subscriber_test.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SubscriberTest : public rclcpp::Node
{
    public:
    SubscriberTest() : Node("subscriber_test") 
    {
        // Placeholder indica cuantos argumentos recibe la función de callback, en caso de haber dos
        // seria lo mismo que hay + ", std::placeholders::_2, std::placeholder::_3..."
        m_subscriber = this->create_subscription<example_interfaces::msg::String>("robot_news", 10,
                            std::bind(&SubscriberTest::subscriberCallback, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Subscriber started");
    }
    private:
        void subscriberCallback(const example_interfaces::msg::String::SharedPtr msg)
        {
            RCLCPP_INFO(this->get_logger(), "Received data!");
            RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
        }
        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr m_subscriber;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SubscriberTest>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_first_package/my_first_package/tests_topics/subscriber_test.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class SubscriberTest(Node):
    def __init__(self, node_name):
        super().__init__(node_name)
        # Parámetros = tipo de dato, nombre topic, función callback para recoger el msg, queue size
        self.subscriber_ = self.create_subscription(String, "robot_news", self.callback_robot_news, 10)

    def callback_robot_news(self, msg: String):
        self.get_logger().info(msg.data)
def main(args=None):
    rclpy.init(args=args)
    node = SubscriberTest("my_first_node")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()  